plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -688,17 +680,60 @@ function( $key ) use ( $order_fields_keys ) {
 	}
 
 	/**
-	 * Validates a field value for a given group.
+	 * Validates a set of fields for a given location against custom validation rules.
+	 *
+	 * @param array  $fields Array of key value pairs of field values to validate.
+	 * @param string $location The location being validated (address|contact|additional).
+	 * @param string $group The group to get the field value for (shipping|billing|'') in which '' refers to the additional group.
+	 * @return WP_Error
+	 */
+	public function validate_fields_for_location( $fields, $location, $group = '' ) {
+		$errors = new WP_Error();
+
+		try {
+			/**
+			 * Pass an error object to allow validation of an additional field.
+			 *
+			 * @param WP_Error $errors  A WP_Error object that extensions may add errors to.
+			 * @param mixed    $fields  List of fields (key value pairs) in this location.
+			 * @param string   $group   The group of this location (shipping|billing|'').
+			 *
+			 * @since 8.7.0
+			 */
+			do_action( 'woocommerce_blocks_validate_' . $location . '_fields', $errors, $fields, $group );
```

Comment:
When the `additional` area is validated, the hook is:
`woocommerce_blocks_validate_additional_fields` which may be easily confused with `woocommerce_blocks_validate_additional_field`. Do you think this is an issue? Should we consider renaming one or the other? Maybe adding `location` to the location hook would work? `woocommerce_blocks_validate_location_additional_fields`

Reply:
Added in https://github.com/woocommerce/woocommerce/pull/44463/commits/b42414966ced846d1166f8874a66a5f37a73be50